### PR TITLE
add profile task query segment list and analyzation

### PR DESCRIPTION
**Pull request by mrproliu**

### CODE REVIEW

File: profile.graphqls [hunk 1]
```diff
@@ -92,4 +112,8 @@ extend type Mutation {
 extend type Query {
     # query all task list, order by ProfileTask#startTime descending
     getProfileTaskList(serviceId: ID, endpointName: String): [ProfileTask!]!
+    # query all task profiled segment list
+    getProfileTaskSegmentList(taskID: String): [BasicTrace!]!
+    # analyze profiled segment
+    getProfileAnalyze(segmentId: String!, duration: Duration!): ProfileAnalyzation!
```
Comment (wu-sheng) [inline profile.graphqls]: We should not use Duration here. As you can see, this ProfileAnalyzation is targeting a specific time range in milliseconds always.

File: profile.graphqls [hunk 2]
```diff
@@ -84,6 +84,26 @@ type ProfileTask {
     logs: [ProfileTaskLog!]!
 }
 
+# Profile thread stack anayze tree
+type ThreadStackAnalyzation {
+    # stack code signature
+    codeSign: String!
```
Comment (wu-sheng) [inline profile.graphqls]: `Sign` is a different word, use `signature`.

File: profile.graphqls [hunk 3]
```diff
@@ -84,6 +84,26 @@ type ProfileTask {
...
+    # self include children duration(millisecond)
+    duration: Int!
+    # self exclude children duration(millisecond)
+    excludeChildDuration: Int!
```
Comment (wu-sheng) [inline profile.graphqls]: `excludeChildDuration` -> `durationExcludeChild`.

File: profile.graphqls [hunk 4]
```diff
@@ -84,6 +84,26 @@ type ProfileTask {
...
+    # total dump count
+    count: Int!
+    # children of this stack code sign
+    childs: [ThreadStackAnalyzation!]
+}
+
+# Profile analyze result
+type ProfileAnalyzation {
```
Comment (wu-sheng) [inline profile.graphqls]: Are you considering this because of multiple roots?

**Comment by wu-sheng** [comment]
After this merged, recommend you to send a separate pull request and just implement the shell of this protocol, make compiling and boot successfully. Because otherwise, it may block other PRs.

File: profile.graphqls (hunk 4 above)
Comment (mrproliu) [reply profile.graphqls]: Yes, It's work for multiple roots encase.

**Comment by mrproliu** [comment]
> After this merged, recommend you to send a separate pull request and just implement the shell of this protocol, make compiling and boot successfully. Because otherwise, it may block other PRs.

Sure.

File: profile.graphqls [hunk 5]
```diff
@@ -84,6 +84,28 @@ type ProfileTask {
...
+    # stack code signature
+    codeSignature: String!
+    # self include children duration(millisecond)
+    duration: Int!
+    # self exclude children duration(millisecond)
+    durationChildExcluded: Int!
+    # total dump count
+    count: Int!
+    # max recursion invoke count
+    maxRecursionCount: Int!
```
Comment (wu-sheng) [inline profile.graphqls]: Don't recommend use this. Stack depth is limited in the agent. We could say it is safe here.

File: profile.graphqls [hunk 6]
```diff
@@ -84,6 +84,28 @@ type ProfileTask {
     logs: [ProfileTaskLog!]!
 }
 
+# Profile thread stack anayze tree
+type ThreadStackAnalyzation {
```
Comment (wu-sheng) [inline profile.graphqls]: `ProfilingStackElement`

File: profile.graphqls [hunk 7]
```diff
@@ -92,4 +112,8 @@ extend type Mutation {
 extend type Query {
     # query all task list, order by ProfileTask#startTime descending
     getProfileTaskList(serviceId: ID, endpointName: String): [ProfileTask!]!
+    # query all task profiled segment list
+    getProfileTaskSegmentList(taskID: String): [BasicTrace!]!
```
Comment (wu-sheng) [inline profile.graphqls]: BasicTrace is in `trace.graphqls`. You need to import it.

**Comment by wu-sheng** [comment]
Please provide the implementation and update the submodule in the main repo.

**Pull request merged**